vignettes/advanced.Rmd

```diff
@@ -0,0 +1,128 @@
+---
+title: "Advanced usage"
+author: "Patrick Schratz, Kirill Müller"
+date: "`r Sys.Date()`"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Kirill Müller}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r setup, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+# Deployment
+
+## pkgdown deployment
+
+You may optionally set the deployment to the `gh-pages` branch and then change the branch serving your site in the Github repo settings (change it to `gh-pages`).
+By this you do not have all the commits from Travis in your `master` branch but your `pkgdown` site will be silently updated in the backgroud without any disturbance.
+Simply set the `branch` argument to `gh-pages` and `orphan = TRUE` so that a force git push is initiated.
+Now, `step_push_deploy(branch = "gh-pages", orphan = TRUE)` will deploy the resulting files of your `pkgdown` site to the `gh-pages` branch of your repo.
+
+## Committing single files
+
+Function `step_push_deploy()` has the ability to only commit and push single files if they have changed during a CI build.
+This can be very useful for conditionally pushing documentation files like `NEWS` or `man/` and `NAMESPACE` if these are automatically created via Travis.
+Simply add the desired files to argument `commit_paths` in `step_push_deploy()`.
+
+# Running stages conditionally
+
+If you run multiple Travis builds on every push (e.g. if you test on multiple R versions or if you are using "build stages" on Travis (i.e. multiple builds that start sequentially after each other)), you often need to condition the `tic` stages based on environment variables.
+The most common use case is the testing on multiple R versions.
+
+Usually, `tic` initiates a code coverage testing in the `after_success` stage (see `?add_package_checks()`).
+Now you do not want that this is run on every R version as the outcome will always be the same. 
+Furthermore, it might trigger problems when all coverages are pushed at roughly the same time.
```

Comment:
I think _codecov_ can distinguish different Travis/AppVeyor builds, but resource conservation is a good argument given that it's really unlikely that coverage differs between R versions. (If it does, it might indicate a serious problem, though.)

Reply:
> Furthermore, it might trigger problems when all coverages are pushed at roughly the same time.

Yeah, a somewhat weak statement. I removed it and made the resource argument stronger.